pkg/remote/aws/appautoscaling_scheduled_action_enumerator.go

```diff
@@ -0,0 +1,59 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/applicationautoscaling"
+	"github.com/cloudskiff/driftctl/pkg/remote/aws/repository"
+	remoteerror "github.com/cloudskiff/driftctl/pkg/remote/error"
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+)
+
+type AppAutoscalingScheduledActionEnumerator struct {
+	repository repository.AppAutoScalingRepository
+	factory    resource.ResourceFactory
+}
+
+func NewAppAutoscalingScheduledActionEnumerator(repository repository.AppAutoScalingRepository, factory resource.ResourceFactory) *AppAutoscalingScheduledActionEnumerator {
+	return &AppAutoscalingScheduledActionEnumerator{
+		repository,
+		factory,
+	}
+}
+
+func (e *AppAutoscalingScheduledActionEnumerator) SupportedType() resource.ResourceType {
+	return aws.AwsAppAutoscalingScheduledActionResourceType
+}
+
+func (e *AppAutoscalingScheduledActionEnumerator) Enumerate() ([]*resource.Resource, error) {
+	actions := make([]*applicationautoscaling.ScheduledAction, 0)
+
+	for _, ns := range e.repository.ServiceNamespaceValues() {
+		results, err := e.repository.DescribeScheduledActions(ns)
+		if err != nil {
+			return nil, remoteerror.NewResourceListingError(err, string(e.SupportedType()))
+		}
+		actions = append(actions, results...)
+	}
+
+	results := make([]*resource.Resource, len(actions))
+
+	for _, action := range actions {
```

Comment:
Why are you doing two loops here ? Can't we create the resource in the first iteration directly ?

Reply:
No because each namespace can embed several scheduled actions (see line 37)

Reply:
Yeah but we can create resources directly in the first loop I think, there is no need to iterate twice.

```go
for _, ns := range e.repository.ServiceNamespaceValues() {
		results, err := e.repository.DescribeScheduledActions(ns)
		if err != nil {
			return nil, remoteerror.NewResourceListingError(err, string(e.SupportedType()))
		}
		for _, result := range results {
                    e.factory.CreateAbstractResource ...
               }
	}
```

Reply:
I think two loops makes things easier to read but I can change that if you want

Reply:
This is really minor so as you want, but at least change `results` naming of `e.repository.DescribeScheduledActions` it can be a little bit confusing IMHO